### PR TITLE
ci: rename release branches

### PR DESCRIPTION
**Pull request by rabelloo**
## Purpose

Allow branches like `release/1.0.1` to be able to publish too.

## Approach

Rename `release/*.x.x` to `release/*` in all CI related files.

## Testing

Check CI after merge when a new `release/*` branch is created.

## Risks

None, already changed GitHub branch settings.

### CODE REVIEW

**Comment by josokinas** [comment]
What do you think if release docs are also updated to:

```md
### Release the latest version (example)

You will need `release/*` branch push permissions.

In the following example we assume that:

- the latest version is 1.0.0
- the version 1.1.0 is being released

```sh
# ensure tree is clean - WARNING: will delete pending changes
git reset --hard

# get all latest changes from remote
git fetch --all --prune

# checkout latest 'main'
git checkout main
git pull

# create and checkout release branch
git checkout -b release/1.1.0

# use automated release script
npm run release

# push to trigger 'publish' GitHub Action
git push --follow-tags origin release/1.1.0
```

Open a release PR to merge the version bump and the changelog back to `main` branch.

```sh
# once merged, checkout merge commit on 'main'
git checkout c6da29b

# move version tag
git tag v1.1.0 -f
git push origin v1.1.0 -f
```

### Release a non-latest minor/patch version (example)

You will need `release/*` branch push permissions.

In the following example we assume that:

- the latest version is 2.0.1
- the version 1.2.1 is being released
- we'll cherry-pick hotfixes from 2.0.1 onto 1.2.0

```sh
# ensure tree is clean - WARNING: will delete pending changes
git reset --hard

# get all latest changes from remote
git fetch --all --prune

# checkout latest available version tag for v1.x.x
git checkout v1.2.0

# create and checkout release branch
git checkout -b release/1.2.x

# cherry-pick required fixes from 'main'
git cherry-pick 64b6be1
# pick appropriate commit hashes and repeat as needed

# use automated release script
npm run release

# push to trigger 'publish' GitHub Action
git push --follow-tags -u origin release/1.2.x
```

No need to open a PR to merge a non-latest release back to `main`, nor do tags need to be moved.
```

On this repo, then also on [Castor Icons](https://github.com/onfido/castor-icons/pull/88).

**Comment by rabelloo** [comment]
> What do you think if release docs on this repo and on [Castor Icons](https://github.com/onfido/castor-icons/pull/88) are also updated to:

It was hard to parse the difference until I applied it locally to my file, but yea I think we should update it.
New commit has the changes.

**Review by josokinas — APPROVED** [review]
